pkg/action/scan.go

```diff
@@ -36,8 +36,11 @@ var (
 	// compiledRuleCache are a cache of previously compiled rules.
 	compiledRuleCache atomic.Pointer[yarax.Rules]
 	// compileOnce ensures that we compile rules only once even across threads.
-	compileOnce         sync.Once
-	ErrMatchedCondition = errors.New("matched exit criteria")
+	compileOnce           sync.Once
+	ErrMatchedCondition   = errors.New("matched exit criteria")
+	ErrFileScan           = NewFileReportError(nil, "", "scan failed")
+	ErrGenerateFileReport = NewFileReportError(nil, "", "failed to generate file report")
+	ErrNilFileReport      = NewFileReportError(nil, "", "file report is nil")
```

Comment:
Move these into scan_error.go and use the consts to make sure that these can't diverge.

Reply:
Actually, are these used anywhere?

Reply:
I may have duped while moving things around. I'll clean these up 👍🏻

Reply:
Updated in [`8e1e5c6` (#777)](https://github.com/chainguard-dev/malcontent/pull/777/commits/8e1e5c6cf378a547f46bccdd7d73ceb9a912eb5b).